app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
   belongs_to :user, foreign_key: :author_id
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts_counter + 1)
+  end
+
```

Comment:
- [ ] It would be better if you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private :blush: